extensions/oplogPopulator/OplogPopulatorMetrics.js

```diff
@@ -0,0 +1,164 @@
+const { ZenkoMetrics } = require('arsenal').metrics;
+
+const { getStringSizeInBytes } = require('../../lib/util/buffer');
+
+class OplogPopulatorMetrics {
+    /**
+     * @param {Logger} logger logger instance
+     */
+    constructor(logger) {
+        this.acknowledgementLag = null;
+        this.connectorConfiguration = null;
+        this.requestSize = null;
+        this.connectors = null;
+        this.reconfigurationLag = null;
+        this._logger = logger;
+    }
+
+    registerMetrics() {
+        this.acknowledgementLag = ZenkoMetrics.createHistogram({
+            name: 'oplog_populator_acknowledgement_lag_sec',
+            help: 'Delta between a config change in mongo and it getting processed by the oplogPopulator in seconds',
+            labelNames: ['opType'],
+            buckets: [0.001, 0.01, 1, 10, 100, 1000, 10000],
+        });
+        this.connectorConfiguration = ZenkoMetrics.createCounter({
+            name: 'oplog_populator_connector_configuration',
+            help: 'Configurations applied to connectors',
+            labelNames: ['connector', 'opType'],
+        });
+        this.buckets = ZenkoMetrics.createGauge({
+            name: 'oplog_populator_connector_buckets',
+            help: 'Number of buckets per connector',
+            labelNames: ['connector'],
+        });
+        this.requestSize = ZenkoMetrics.createCounter({
+            name: 'oplog_populator_connector_request_bytes',
+            help: 'Size of kafka connect request in bytes',
+            labelNames: ['connector'],
+        });
+        this.mongoPipelineSize = ZenkoMetrics.createGauge({
+            name: 'oplog_populator_connector_pipeline_bytes',
+            help: 'Size of mongo pipeline in bytes',
+            labelNames: ['connector'],
+        });
+        this.connectors = ZenkoMetrics.createGauge({
+            name: 'oplog_populator_connectors',
+            help: 'Total number of configured connectors',
+            labelNames: ['isOld'],
+        });
+        this.reconfigurationLag = ZenkoMetrics.createHistogram({
+            name: 'oplog_populator_reconfiguration_lag_sec',
+            help: 'Time it takes to reconfigure a connector',
```

Comment:
is it the time it takes to configure a _single_ connector? (e.g. issuing the command to kafka connect)
or does it also include the time it takes for a _single_ connector to "be ready" (e.g. tasks re-created/updated in kafka connect)?
or the overall "reconfigure time" (that it takes to reconfigure all connectors)